db/DB.js

```diff
@@ -419,6 +419,30 @@ const DB = {
     await knex('email_addresses').where({ subscriber_id: uid }).del()
   },
 
+  async updateBreachStats (id, stats) {
+    await knex('subscribers')
+      .where('id', id)
+      .update({
+        breach_stats: stats
+      })
+  },
+
+  async getBreachStats (id) {
+    const { breach_stats: breachStats } = await knex('subscribers')
+      .where({ id })
+      .first('breach_stats')
+
+    return breachStats
+  },
+
+  async getSubscribersWithUnresolvedBreaches () {
+    const res = await knex('subscribers')
+      .select('primary_email', 'breach_stats', 'signup_language')
+      .whereJsonPath('breach_stats', '$.numBreaches.numUnresolved', '>', 0)
```

Comment:
risk: I worry about how this `WHERE breach_stats ...` query will perform at the scale of thousands or millions of records. (E.g., the [4.64M `en` subscribers](https://earthangel-b40313e5.influxcloud.net/d/dEpkGp4Wz/fx-monitor?orgId=1&viewPanel=45)) There's no index on the field, and I'm not 100% sure how indexes work on `jsonb` fields.

If this query only returns subscribers who have non-null value in `breach_stats` - i.e., subscribers who have visited Monitor and thereby populated their `breach_stats` field, then it seems fine For Now™ so long as we can monitor its DB performance to make sure it's not getting too wild as more and more subscribers are included in it.

Reply:
If we're punting the cron job to SRE, we could have them monitor it as well... 😄